http/http-advanced-reactive/src/test/java/io/quarkus/ts/http/advanced/reactive/Brotli4JHttpIT.java

```diff
@@ -0,0 +1,160 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.restassured.response.Response;
+import jakarta.ws.rs.core.HttpHeaders;
+import jakarta.ws.rs.core.MediaType;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import io.quarkus.test.bootstrap.RestService;
+import io.quarkus.test.scenarios.QuarkusScenario;
+import io.quarkus.test.services.QuarkusApplication;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Scanner;
+
+@Tag("QQE-378")
+@QuarkusScenario
+public class Brotli4JHttpIT {
+    @QuarkusApplication(classes = { Brotli4JHttpServerConfig.class, Brotli4JResource.class, Brotli4JRestMock.class
+    }, properties = "compression.properties")
+    static RestService app = new RestService();
+
+    private final static String DEFAULT_TEXT_PLAIN = "In life, you have to trust that every little bit helps. As you know," +
+            " every small step forward counts." +
+            " It's the accumulation of these efforts that ultimately leads to success." +
+            " So, don't underestimate the power of persistence and determination in achieving your dreams";
+
+    private final static int CONTENT_LENGTH_DEFAULT_TEXT_PLAIN = DEFAULT_TEXT_PLAIN.length();
+
+    private final static String JSON_CONTENT= "Hello from a JSON sample";
+
+    private final static String BROTLI_ENCODING = "br";
+
+    @Test
+    public void checkTextPlainWithoutBrotli4JEncoding() {
+        app.given()
+                .get("/compression/text")
+                .then()
+                .statusCode(200)
+                .and()
+                .header(HttpHeaders.CONTENT_LENGTH, String.valueOf(CONTENT_LENGTH_DEFAULT_TEXT_PLAIN))
+                .header(HttpHeaders.CONTENT_ENCODING, nullValue())
+                .body(is(DEFAULT_TEXT_PLAIN)).log().all();
+    }
+
+    @Test
+    public void checkTextPlainWithtBrotli4J() throws IOException {
+        assertBrotli4JCompression("/compression/text", MediaType.TEXT_PLAIN, BROTLI_ENCODING, BROTLI_ENCODING, CONTENT_LENGTH_DEFAULT_TEXT_PLAIN);
+
+    }
+
+    public void assertBrotli4JCompression(String path, String contentHeaderType, String acceptHeaderEncoding, String expectedHeaderContentEncoding, int originalContentLength) {
+       Response response = app.given()
+                .when()
+                .contentType(contentHeaderType)
+                .header(HttpHeaders.ACCEPT_ENCODING, acceptHeaderEncoding)
+                .get(path)
+                .then()
+                .statusCode(200)
+                .header(HttpHeaders.CONTENT_ENCODING, expectedHeaderContentEncoding)
+               .extract().response();
+        byte[] responseBody = response.getBody().asByteArray();
+        int compressedContentLength = responseBody.length;
+        assertTrue(compressedContentLength < originalContentLength);
+
+    }
+
+    @Test
+    public void disableCompression(){
+        System.setProperty("quarkus.http.enable-compression", "false");
```

Comment:
I'm not sure if this is correct usage with quarkus. Maybe it is but can't say what should be correct way.

On another note, how we ensuring that this property is only set for this test and not affecting others test?

Also I tried this test, removed this property and the test still pass. Shouldn't this fail when compression is disabled?

Reply:
Yeah, good point @jedla97 and thank you for the explanation @michalvavrik .
I will fix it, sorry about that guys.

Reply:
done with @DevModeQuarkusApplication